numpy/core/_exceptions.py

```diff
@@ -121,3 +121,17 @@ def __init__(self, axis, ndim=None, msg_prefix=None):
                 msg = "{}: {}".format(msg_prefix, msg)
 
         super(AxisError, self).__init__(msg)
+
+
+@_display_as_base
+class _ArrayMemoryError(MemoryError):
+    """ Thrown when an array cannot be allocated"""
+    def __init__(self, shape, dtype):
+        self.shape = shape
+        self.dtype = dtype
+
+    def __str__(self):
+        return "Attempted to allocate array with shape " + \
+        str(self.shape).rstrip(')').rstrip(',') + ") and data type " + \
+        str(self.dtype) + "."
```

Comment:
I don't think you should remove the trailing comma from the `shape`.

Also, consider using `.format` here, as
```python
return "Unable to allocate array with shape {} and data type {}".format(self.shape, self.dtype)
```